### PR TITLE
Add questionExamBlock environment

### PR DESCRIPTION
**Pull request by ebousse**
A problem with the commands `\questionExam` and `\answerExam` is that they may lead to orphan answer boxes separated from questions, eg. like this:

```latex
\questionExam[q.mylabel]{\lipsum[1][3]}

% You can use the question's label like this:
\textbf{Reference to \cref{q.mylabel}}
% Warning: there is an issue when using question's label and section/sub-section.

\answerExam[white]{1.5}{8}{1}
```

Capture d’écran du 2024-03-28 15-17-15

This PR proposes a new environment called `questionExamBlock` which works the same as `questionExam`, but makes sure all content is in a minipage that ensures everything will always be shown together.

Example:
```latex
\begin{questionExamBlock}[q.mylabel]{\lipsum[1][3]}

   % You can use the question's label like this:
   \textbf{Reference to \cref{q.mylabel}}
   % Warning: there is an issue when using question's label and section/sub-section.

   \answerExam[white]{1.5}{8}{1}

\end{questionExamBlock}
```

Result:

Capture d’écran du 2024-03-28 15-17-57

### CODE REVIEW

**Comment by arnobl** [comment]
Thank you again.
Could you please add a commented example of the new environment in the demonstration file:  `exam-example.tex`?

**Comment by ebousse** [comment]
Added a fix (woopsies) and added the requested example.

I think I found another issue with my code… wait until merge.

**Review by arnobl — APPROVED** [review]

**Comment by arnobl** [comment]
Thank you!

**Comment by ebousse** [comment]
I think you did not read the last part of my last message :sweat_smile: 

I think the environment creates strange corner cases with labels and references, I'm investigating and I'll add another PR if I fix it. I hadn't found this before since we were not using labels in our exam.

**Comment by arnobl** [comment]
There is a bug with labels I was not able to fix yet: if the labelled question is the first one of the exam, its references will not work.

**Pull request merged**
